tests/integration/device_test.sh

```diff
@@ -16,67 +16,39 @@ function device_info_test_helper()
-  container="kw-${distro}"
-  buffer=$(container_exec "${container}" 'kw device --local')
+  for distro in "${DISTROS[@]}"; do
+    container="kw-${distro}"
 
-  # some fields must be ignored because they surely won't match.
-  filter=(
-    'Root filesystem' 'Size' 'Mounted on'                                            # storage fields
-    'Distribution' 'Distribution base' 'Distribution version' 'Desktop environments' # desktop fields
-  )
+    container_tmp_dir=$(container_exec "$container" 'mktemp --directory')
 
-  # add | separator to each filter item.
-  filter_regex=$(sed 's/ /|/g' <<< "${filter[@]}")
+    # Copy the kw_device_expected.sh to the container's workdir directory
+    container_copy "$container" "$kw_device_simulation" "$container_tmp_dir"
 
-  # the actual perl regex.
-  filter_regex="(${filter_regex}):"
+    # Get the output of the `kw device --local` command from the container
+    buffer=$(container_exec "$container" 'kw device --local')
 
-  # get the output, after applying filter.
-  output=$(printf '%s' "${buffer}" | grep --invert-match --perl-regexp "${filter_regex}")
+    # Specify the fields to filter out from the output
+    filter=(
+      'Distribution base' 'Distribution version' 'Desktop environments' # desktop fields
+    )
 
-  # get the expected output from the host machine, also filtering information.
-  expected_output=$(device_main --local | grep --invert-match --perl-regexp "${filter_regex}")
+    # Construct a regex by joining the filter items with a | separator
+    filter_regex=$(printf '%s|' "${filter[@]}" | sed 's/|$//')
 
-  # TODO: remove this block of code after issue #1000 has been solved.
-  #
-  # deviceinfo shows GPU information for Arch, but not for Debian and Fedora due
-  # some libraries being present in Arch by default,  but  not  in  the  others.
-  # Therefore, if this information is not present, we must filter it out.
-  grep 'GPU:' <<< "${output}" > /dev/null
-  if [[ "$?" -ne 0 ]]; then
-    # we resort to SED because GREP can't filter out lines after context IF we
-    # use invert match. First, we tell sed to stop printing lines after matching
-    # the pattern GPU then delete the line containing that pattern.
-    expected_output=$(sed '/GPU:/q' <<< "${expected_output}" | sed '/GPU:/d')
-  fi
+    # The actual perl regex.
+    filter_regex="(${filter_regex}):"
 
-  assertEquals "(${LINENO}): kw device failed for ${distro}" "${expected_output}" "${output}"
+    # Get the output, after applying filter.
+    output=$(printf '%s' "${buffer}" | grep --invert-match --perl-regexp "$filter_regex")
 
-  # check storage information
-  fs_type=$(container_inspect --format '{{.Driver}}' "${container}")
-  fs_size=$(container_inspect --format '{{.GraphDriver.Data.WorkDir}}' "${container}" |
-    xargs df -h | tail --lines 1 | awk '{ print $2 }')
+    # Get the expected output
+    expected_output=$(container_exec "$container" "bash ${container_tmp_dir}/kw_device_expected_output.sh")
```

Comment:
tbh, I did not like this approach since it creates a script similar to the device that we have to maintain. How about changing how we approach this problem a little bit? We can try something like this:
1. Run kw device in the container and capture the output.
2. Create many small tests that check just part of the output. 

For example, you can create one test just for validate the gpu return; for that, you just run the command to retrive the gpu output from the container and compare it with the result from `kw device`.